packages/vega-parser/src/parsers/guides/axis-grid.js

```diff
@@ -38,49 +40,80 @@ export default function(spec, config, userEncode, dataRef, band) {
     round:  _('tickRound')
   };
 
-  if (orient === Top || orient === Bottom) {
-    u = 'x';
-    v = 'y';
-    s = 'height';
-  } else {
-    u = 'y';
-    v = 'x';
-    s = 'width';
-  }
-  v2 = v + '2';
-
-  update[u] = enter[u] = exit[u] = tickPos;
-
-  if (vscale) {
-    update[v] = enter[v] = {scale: vscale, range: 0, mult: sign, offset: offset};
-    update[v2] = enter[v2] = {scale: vscale, range: 1, mult: sign, offset: offset};
+  if (!isSignal(orient)) {
+    if (orient === Top || orient === Bottom) {
+      u = 'x';
+      v = 'y';
+      s = 'height';
+    } else {
+      u = 'y';
+      v = 'x';
+      s = 'width';
+    }
+    v2 = v + '2';
+  
+    update[u] = enter[u] = exit[u] = tickPos;
+  
+    if (vscale) {
+      update[v] = enter[v] = {scale: vscale, range: 0, mult: sign, offset: offset};
+      update[v2] = enter[v2] = {scale: vscale, range: 1, mult: sign, offset: offset};
+    } else {
+      update[v] = enter[v] = {value: 0, offset: offset};
+      update[v2] = enter[v2] = {signal: s, mult: sign, offset: offset};
+    }
   } else {
-    update[v] = enter[v] = {value: 0, offset: offset};
-    update[v2] = enter[v2] = {signal: s, mult: sign, offset: offset};
+    update.x = enter.x = xAxisConditionalEncoding(orient.signal, tickPos, vscale ? { scale: vscale, range: 0, mult: sign, offset: offset} : { value: 0, offset: offset});
```

Comment:
This part is really hard to read, and repeating the same logic with ternary operators logic multiple times

- `vscale ? { scale: vscale, range: 0, mult: sign, offset: offset} : { value: 0, offset: offset}`

- `vscale ? {scale: vscale, range: 1, mult: sign, offset: offset} : {signal: xAxisExpr(orient.signal, 'height', 'width', false, false).signal, mult: sign, offset: offset}`

Extract common part and name them so we know what they are?

Reply:
yep, done.